lib/scanner/walk.go

```diff
@@ -218,18 +256,23 @@ func (w *walker) walkAndHashFiles(ctx context.Context, fchan, dchan chan protoco
 			skip = fs.SkipDir
 		}
 
-		if err != nil {
-			l.Debugln("error:", path, info, err)
-			return skip
-		}
-
-		if path == "." {
-			return nil
+		if err == nil {
```

Comment:
This stuff could just be below the `if err != nil` below?

Reply:
Inside this there `err` can become non-nil in 
```
info, err = w.Filesystem.Lstat(path)
```

This was present before this PR however the fs walker already does a Lstat. I assumed this has a point that I just don't see, but if there isn't, I can remove it and which will clean this section up a bit (tests pass without this additional LStat).